packages/cli/lib/validators/marketplaceValidators/theme/SectionValidator.js

```diff
@@ -0,0 +1,116 @@
+const {
+  ANNOTATION_KEYS,
+  getAnnotationValue,
+  getFileAnnotations,
+} = require('@hubspot/cli-lib/templates');
+const BaseValidator = require('../BaseValidator');
+const { VALIDATOR_KEYS } = require('../../constants');
+
+const SECTION_LIMIT = 50;
+
+class SectionValidator extends BaseValidator {
+  constructor(options) {
+    super(options);
+
+    this.errors = {
+      LIMIT_EXCEEDED: {
+        key: 'limitExceeded',
+        getCopy: ({ limit, total }) =>
+          `Section limit exceeded. Themes can only have ${limit} sections, but this theme has ${total}`,
+      },
+      MISSING_LABEL: {
+        key: 'missingLabel',
+        getCopy: ({ filePath }) =>
+          `Missing required property for ${filePath}. The section is missing the "label" property`,
+      },
+      MISSING_SCREENSHOT_PATH: {
+        key: 'missingScreenshotPath',
+        getCopy: ({ filePath }) =>
+          `Missing required property for ${filePath}. The section is missing the "screenshotPath" property`,
+      },
+      MISSING_DESCRIPTION: {
+        key: 'missingDescription',
+        getCopy: ({ filePath }) =>
+          `Missing required property for ${filePath}. The section is missing the "description" property`,
+      },
+    };
+  }
+
+  // Validates:
+  // - All sections have a 'description' annotation
+  // - All sections have a "label" annotation
+  // - All sections have a "screenshotPath" annotation
+  // - Theme does not have more than SECTION_LIMIT sections
+
+  validate(files) {
+    let validationErrors = [];
+    let sectionCount = 0;
+
+    files.forEach(file => {
+      if (file) {
+        const annotations = getFileAnnotations(file);
+        const templateType = getAnnotationValue(
+          annotations,
+          ANNOTATION_KEYS.templateType
+        );
+        const isAvailableForNewContent = getAnnotationValue(
+          annotations,
+          ANNOTATION_KEYS.isAvailableForNewContent
+        );
+
+        if (templateType !== 'section') {
+          return;
+        }
+        sectionCount++;
+
+        if (isAvailableForNewContent !== 'false') {
```

Comment:
Just confirming: sections with `isAvailableForNewContent === false` do not could towards the limit?

Reply:
Correct. I am open to discussion on this one, but I feel that we can allow developers to use sections as building blocks without offering them to customers.
@KatzMitch curious your thoughts here, and how you're approaching the internal validation.

Reply:
I vote we keep it the same as templates - looks like it's 50 total, not 50 avail for new content (https://github.com/HubSpot/hubspot-cli/blob/master/packages/cli/lib/validators/marketplaceValidators/theme/TemplateValidator.js#L124-L125)
Also do we need to update the ignored type list in that file?

Reply:
> I vote we keep it the same as templates - looks like it's 50 total, not 50 avail for new content

Sounds good! I'll account for that.

> Also do we need to update the ignored type list in that file?

Yep! Accounting for that - https://github.com/HubSpot/hubspot-cli/blob/marketplace/section-support/packages/cli/lib/validators/marketplaceValidators/theme/TemplateValidator.js#L12

Reply:
@KatzMitch @anthmatic updates to account for non-available sections in https://github.com/HubSpot/hubspot-cli/pull/546/commits/a36ec9f9cf6d78a3519e10c4e3c41ed3e7cb074f